### PR TITLE
[Microbyte] IPv4 Addressing

### LINKED ISSUE #11: IPv4 addressing
* MicroByte Title: IPv4 addressing

* Domain: Networking

* Scope of learning for learners:
Learning about classfull and classless IP addressing

* Activities:
Conversion from dotted decimal format to binary and vice-versa, finding network mask,identifying different classes of IP addresses etc

### PR DESCRIPTION
**Pull request by Anjalikamath**
## Closes #11 
Added a microbyte to learn some terminologies in IPv4 addressing
 Checklist: 
✅ Added activities for learners
✅ Didn't give the solutions to the added activities straight away 
✅ Followed a LearnByDoing approach 
✅ Added images for better understanding if deemed necessary

### CODE REVIEW

**Comment by YourAKShaw** [comment]
@Anjalikamath 

Kindly follow the contribution guidelines as mentioned in the [CONTRIBUTING.md](https://github.com/Crio-Bytes/Networking/blob/main/CONTRIBUTING.md) file of the repo. 

- According to our contribution guidelines, you are required to create an issue for a new MicroByte Proposal, similar to https://github.com/Crio-Bytes/Miscellaneous/issues/3. 
- You are required to utilize the `Add MicroByte` issue template for the same. 
- You need not wait for yourself to be assigned for your created issue, and can start working right away.
- Once, your work is done, you need to create a PR similar to https://github.com/Crio-Bytes/Miscellaneous/pull/1.
- You are requested to utilize [this](https://github.com/Crio-Bytes/Networking/blob/main/.github/PULL_REQUEST_TEMPLATE/adding-microbyte.md) PR template for the same.
- For further clarification, you can also go ahead and watch our YT video: [OSS Contribution Workflow of microbytes](https://www.youtube.com/watch?v=qdSn8v9nU0g&feature=youtu.be).

**Comment by YourAKShaw** [comment]
Kindly create the required issue and edit your PR description.

Also, your PR is currently under review.

**Comment by amoghrajesh** [comment]
Hey @Anjalikamath!
You need to open an issue pertaining to this PR by following an issue template as provided in .github folder of each repo. 
Post that, you need to create this particular PR details using a PR template provided in the same folder.
We will have a look asap after an issue is created and the PR details follow a template too 😄

**Comment by Anjalikamath** [comment]
I am so sorry. I'll do the changes.

**Comment by Anjalikamath** [comment]
Closes #11 
Added a microbyte to learn some terminologies in IPv4 addressing
 Checklist: 
✅ Added activities for learners
✅ Didn't give the solutions to the added activities straight away 
✅ Followed a LearnByDoing approach 
✅ Added images for better understanding if deemed necessary

**Comment by YourAKShaw** [comment]
You needn't change the title. You had to edit the description. Do it similar to https://github.com/Crio-Bytes/Miscellaneous/pull/1.

If you have queries. Kindly reach out to us.

**Comment by YourAKShaw** [comment]
> I am so sorry. I'll do the changes.

Chill. We're all here to learn.

**Comment by Kevinpaulose05** [comment]
@Anjalikamath Kindly remove ur solutions from the solution folder from ur microbyte. The reason is after we merge ur MB, u will be **opening issues** for learners to contribute to ur MB. The only way they can do is by answering ur quizzes. So u can open 5 issues and people can contribute into the solutions folder. U don't need to provide them. Else ppl will just use ur MB as a learning material and not as a contributing material.

**Comment by Anjalikamath** [comment]
@Ak-Shaw  I'll do the changes.

**Comment by Anjalikamath** [comment]
@Kevinpaulose05  Okay, I'll update that.

**Comment by Anjalikamath** [comment]
I have updated the microbyte. Kindly let me know if any changes are required.Thanks!

**Comment by YourAKShaw** [comment]
@Anjalikamath I think that you're not getting it. :sweat_smile: 
image
You need to change the description above and use the aforementioned PR template for it.

**Comment by Kevinpaulose05** [comment]
@Anjalikamath Can u pls mention how learners can give the solutions to the quizzes?

My suggestion is:
- Make a demo folder with a demo file called solutions folder like before and indicate abt the folder in the readme itself. And also mention abt this in ur issues.

But this is just a suggestion. feel free to come up with ur own solution. The main idea is to let ppl know how & where they can give the solutions, as that is the most critical aspect of your microbyte.

**Comment by Anjalikamath** [comment]
> @Anjalikamath Can u pls mention how learners can give the solutions to the quizzes?
> 
> My suggestion is:
> 
> * Make a demo folder with a demo file called solutions folder like before and indicate abt the folder in the readme itself. And also mention abt this in ur issues.
> 
> But this is just a suggestion. feel free to come up with ur own solution. The main idea is to let ppl know how & where they can give the solutions, as that is the most critical aspect of your microbyte.

I have changed the folder structure and I have mentioned about this in the Readme as well. Can you please tell me how I can mention about this in my issues? Thanks!

**Comment by Kevinpaulose05** [comment]
> > @Anjalikamath Can u pls mention how learners can give the solutions to the quizzes?
> > My suggestion is:
> > 
> > * Make a demo folder with a demo file called solutions folder like before and indicate abt the folder in the readme itself. And also mention abt this in ur issues.
> > 
> > But this is just a suggestion. feel free to come up with ur own solution. The main idea is to let ppl know how & where they can give the solutions, as that is the most critical aspect of your microbyte.
> 
> I have changed the folder structure and I have mentioned about this in the Readme as well. Can you please tell me how I can mention about this in my issues? Thanks!

Add ur solutions folder that u had initially. Inside it create folders for quizzes like quiz-1,2,3,.. Keep demo files in each(so that while pushing github recognisies the folder). Then wherever u have mentioned "take this quiz", mention "and subkit the solution here" where here points to quiz's solution folder. Then u don't have to mention that seperately in the issue. 

Hope this cleared ur doubt

**Comment by Anjalikamath** [comment]
> > > @Anjalikamath Can u pls mention how learners can give the solutions to the quizzes?
> > > My suggestion is:
> > > 
> > > * Make a demo folder with a demo file called solutions folder like before and indicate abt the folder in the readme itself. And also mention abt this in ur issues.
> > > 
> > > But this is just a suggestion. feel free to come up with ur own solution. The main idea is to let ppl know how & where they can give the solutions, as that is the most critical aspect of your microbyte.
> > 
> > 
> > I have changed the folder structure and I have mentioned about this in the Readme as well. Can you please tell me how I can mention about this in my issues? Thanks!
> 
> Add ur solutions folder that u had initially. Inside it create folders for quizzes like quiz-1,2,3,.. Keep demo files in each(so that while pushing github recognisies the folder). Then wherever u have mentioned "take this quiz", mention "and subkit the solution here" where here points to quiz's solution folder. Then u don't have to mention that seperately in the issue.
> 
> Hope this cleared ur doubt

Thank You so much! This cleared my doubts. I have updated the microbyte.

File: IPV4_Addressing/README.md
Comment (YourAKShaw) [inline IPV4_Addressing/README.md:355]: @Anjalikamath I really appreciate your efforts. This MicroByte has some nice scope for learners. Just one request though. Can you kindly add a few references in the 4 points mentioned in the `FURTHER EXPLORATION` section above.

Other than that, it LGTM! :rocket: 

Let's see what the other reviewers have to say.

**Review by YourAKShaw — CHANGES_REQUESTED** [review]
Kindly make the necessary changes and we'll move forward with reviewing.

**Review by amoghrajesh — APPROVED** [review]
Hey @Anjalikamath!
This is one of the most engaging content I have seen in a while. 
Choosing such a theory related topic and making it seem so interesting is something extraordinary. 
Keep up the good work, no complaints from my end.
Make the changes suggested by @Ak-Shaw and think of some learner issues too
Hoping to see more such work from you 😄

File: IPV4_Addressing/README.md
Comment (Anjalikamath) [reply IPV4_Addressing/README.md:355]: I have modified the file. Thanks!

**Comment by Anjalikamath** [comment]
Thank you so much! @amoghrajesh

**Review by YourAKShaw — APPROVED** [review]
@Anjalikamath Good work. This MicroByte will be really fruitful for learners! :rocket: 

Thank you for your contributions. :heart_decoration:

**Comment by Kevinpaulose05** [comment]
Well done @Anjalikamath . I appreciate ur prompt and perfect responses. Keep up the work. Try coming up with even more microbytes in the following week.

Happy Contributing!!

**Comment by YourAKShaw** [comment]
@Anjalikamath Kindly showcase your MicroByte in the **#microbyte-showcase** channel on slack using [this](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md) template.

**Comment by YourAKShaw** [comment]
@all-contributors kindly add @Anjalikamath for:

- [x] code
- [x] doc
- [x] ideas
- [x] content

**Comment by YourAKShaw** [comment]
@all-contributors kindly add @amoghrajesh for review and eventOrganizing.

**Pull request merged**
